### DISCUSSION TITLE
MikTex install breaks on installing updates

### DISCUSSION THREAD

**Discussion by wolfhound42** [MiKTeX/miktex#1279] [General]
I'm running Windows 10.0.19044.2486.
I deleted my old MikTex installation because it got corrupted. I cleared (I believe) all the entries from the Registry.

I've downloaded and installed a new single-user version. When I try to run the console to update packages, it gets about 8% and then just stops (Task manager shows the process is suspended). When I look in the MikTex\miktex\bin\x64 directory, I see one of the dlls has been renamed: MikTex221000-bzip2.dll.(old).
![image](https://user-images.githubusercontent.com/124470628/216791871-9187d63a-0504-4b84-b92f-ec184622539e.png)

I've reinstalling run the update process several times, but don't know what's happening. Any suggestions for how to fix this would be welcome! Thanks!

**Comment by RonBOakes** [comment]
I've been seeing this today - and I've tried a number of wworkarounds  The only one that has had much success has been install MiKTex on my Mac and continuing my work over there.

It appears that when updating the config package, a key DLL file is getting deleted which causes the process to hang.  My dangerous level of knowledge about the behavior of software (30 years working as a Software Engineer on various projects, but not on many Windows applications at this level) makes me think that the DLL is providing a needed library that is either not in memory or is deleted from memory when the file is deleted.  So Windows hangs the task trying to load the DLL that no longer exists because the task deleted it without replacing it.

The clue comes from attempting to run the MiKTex console again.  It will fail and generate an error about the missing DLL.

**Comment by mlamias** [comment]
I too am experiencing this exact same problem on win 10 x64.  My updates hang and stall at exactly 8%.  It seems during the update process the file C:\Users\Mark\AppData\Local\Programs\MiKTeX\miktex\bin\x64\MiKTeX230100-bzip2.dll is renamed to MiKTeX230100-bzip2.dll(old)and the original dll can't seem to be found which causes the hang.

**Reply by riahimedamine** [reply]
What was the solution/work around? I cannot uninstall miktex the nomral way and the message for the error recommend a reinstallation which just pops the same error message

**Comment by titus483** [comment]
I have the same problem: installation of updates hangs with 8% miktex-config-2.9. Win task manager says process is halted.
So it seems that we can't update MikTeX right now. Re-installation of MikTeX din't help either because it tries to update directly after installation finishes. My bzip2-file (see above) has not been renamed, however.

**Comment by juliaalbers** [comment]
Same Problem here. Updates stucks on 8% and as with @mlamias the file has been renamed with an .OLD. Deleting the .old helped to be able to run the console again. But the Update dont work!

**Comment by lemelinm** [comment]
I have tried to install MIkTex on Win 10 Pro x64 with basic-miktex. I install it as an administrator. When the installation is finished, I am asked if I want to install the updates. When it started, it always gives me an error after a couple of seconds. I have attached the log file. But now, I cannot reinstall it anymore. And I cannot uninstall it because Win does not see it anymore. So I have deleted the folder C:\Users\<<user>>\AppData\Local\Programs\MiKTeX. After that, I try to install it without my antivirus software running (I love living dangerously) but to no avail on success. So now, I am stuck and I cannot continue my work. What should we do?
[MikTex Log error.txt](https://github.com/MiKTeX/miktex/files/10970410/MikTex.Log.error.txt)

**Comment by titus483** [comment]
The 8%-hang on miktek-update seems to have no solution. This bug is open from the beginning of february. Issue reports are closed without any proposal. 
I'm sad to leave an excellent piece of software I used for over 20 years now - but if there is no chance to update it (even after a fresh install) then I have to look for an alternative.
Sorry to say good-bye...

**Reply by lemelinm** [reply]
Do you have a substitute so I can continue to use TexMaker or others?

**Reply by Relevartt** [reply]
Same here. I filed the issue and can't even find it here on Github anymore. Seems like it was deleted, after someone simply said they couldn't reproduce it and closed it. That's what I call terrible support.

**Comment by ASeatonSpatial** [comment]
I get "internal error" when I get to this stage, I don't even get to 8%.  I've tried 3 fresh installs with no luck.  All installs run as administrator on a local user account.  No signs that this is being addressed at all?  

I tried this on a new PC with Windows 10.0.19044, never had miktex installed on it before.

**Reply by ASeatonSpatial** [reply]
Forgot to say, my error report is completely empty.  All I see is GENERAL MIKTEX INFORMATION and nothing else.

**Comment by wolfhound42** [comment]
As I noted on StackExchange on Feb 10, I was able to complete the installation - you might try this: 
"MikTex-Console was suspended in the task manager. Not sure if it made any difference, but I gave the console app permission to reach through the Windows Firewall. Then I went into Windows Task manager, Performance, Open Resource Manager and right clicked to unsuspend MikTex-Console. It resumed and finished with no other issues".

**Comment by Relevartt** [comment]
Same issue here. Starting to look for an alternative to Miktex now because the support here is not what I expected.

**Reply by mlamias** [reply]
Try [TexLive](https://www.tug.org/texlive/).

**Comment by Unai-B** [comment]
I'm getting a similar error, but the update gets stuck at 15% instead. I'm using the 
basic-miktex-23.4-x64.exe installer. Any news?

**Comment by HadisSalasi** [comment]
Hello everyone. I faced a problem when I was trying to update MiKTeX. After selecting the "checked for updates" button, an error message was shown: Couldn't resolve host name, code="6", url="https://ctan.yazd.ac.ir/systems/win32/miktex/tm/packages/miktex-zzdb3-2.9.tar.lzma". And the updating process was not completed. What is your suggestion to fix them? I require the MiKTeX distribution to use the Latex option on VScode. What is another alternative except, Texlive, that helps me? Thanks.

**Comment by szw0407** [comment]
Same issue. Seems somehow an error occurs when it tries to download something, and the whole program just stuck without being able to do anything. It should be a bug. Maybe we should retry or raise an error to allow the user to do something.

Moreover, seems too many update downloads will cause the problem. The problem often happens when downloading over 50 packages, thus might be rare for users who update regularly, but is very harmful to us who just updates everything after a fresh install.

**Comment by szw0407** [comment]
Personally I suggest inspecting how the UI shows the downloading process, in case some internal error keeps it waiting all the time.
